sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -183,6 +185,57 @@ class CollationSuite extends DatasourceV2SQLBase {
     }
   }
 
+  test("aggregates count respects collation") {
+    Seq(
+      ("ucs_basic", Seq("AAA", "aaa"), Seq(Row(1, "AAA"), Row(1, "aaa"))),
```

Comment:
This test case failed in the daily test of Maven + Java 21, both on Linux and MacOS. @dbatomic Do you have time to investigate the cause of this failure?

- linux: https://github.com/apache/spark/actions/runs/8189363924/job/22416511124

![image](https://github.com/apache/spark/assets/1475305/2b39b2d2-a4b2-45ec-8bd9-42c36f22a0b9)


- macos-14: https://github.com/apache/spark/actions/runs/8194082205/job/22416483724

![image](https://github.com/apache/spark/assets/1475305/280c413b-71d6-4715-80a6-1f5ae27a5097)

also cc @HyukjinKwon because I noticed that you are paying attention to the tests of maven + Java 21 on macos-14

Reply:
cc @dongjoon-hyun too

Reply:
Looking into this. Thanks.

Reply:
@stefankandic picked up fixing the issue. This is a race condition in ICU library (problem on spark side where we don't properly setup the library).

Reply:
https://github.com/apache/spark/pull/45436